sidecar/tests/test_native_models.py

```
@@ -685,3 +692,72 @@ def boom(repo, fname, local_files_only=True):
+
+
+# ── byte-level download progress (progress bar for single-GGUF cards) ────────
+
+
+def test_download_reports_byte_progress(monkeypatch, tmp_path):
+    """Single-file cards (every ASR/LLM GGUF) must report BYTES, not file
+    counts — with total = the catalog's size_bytes — so the renderer's bar
+    moves during a multi-GB file instead of sitting at 0/2."""
+    import huggingface_hub
+
+    f1 = tmp_path / "a.gguf"; f1.write_bytes(b"x" * 600)
```

Ruff (0.15.20)

[error] Multiple statements on one line (semicolon)

(E702)